src/boot/efi/console.c

```diff
@@ -132,14 +132,15 @@ EFI_STATUS console_key_read(uint64_t *key, uint64_t timeout_usec) {
 
                 if (FLAGS_SET(keydata.KeyState.KeyShiftState, EFI_SHIFT_STATE_VALID)) {
                         /* Do not distinguish between left and right keys (set both flags). */
-                        if (keydata.KeyState.KeyShiftState & EFI_SHIFT_PRESSED)
-                                shift |= EFI_SHIFT_PRESSED;
                         if (keydata.KeyState.KeyShiftState & EFI_CONTROL_PRESSED)
                                 shift |= EFI_CONTROL_PRESSED;
                         if (keydata.KeyState.KeyShiftState & EFI_ALT_PRESSED)
                                 shift |= EFI_ALT_PRESSED;
                         if (keydata.KeyState.KeyShiftState & EFI_LOGO_PRESSED)
                                 shift |= EFI_LOGO_PRESSED;
+
+                        /* Ideally we would handle shift here too, but that would prevent us from getting
+                         * uppercase letteres on some firmware. */
```

Comment:
i am not sure i follow here? where precisely are the uppercase letters eaten? I mean, at this point you already have the key press info, and then just call KEYPRESS() to merge it, but that's a macro, and firmware is not involved anymore?